### PR TITLE
Box It script

### PR DESCRIPTION
**Pull request by Angadia**
Cohort34 Homework 1 Box It script.
It's the implementation of the optional stretch goal for reading inputs from a CSV file for the executable script from command line. If required, earlier commits on box_it_script branch have implementation of all the required work for Homework 1.

### CODE REVIEW

File: box_it_script/BoxIt.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env node
+
+const drawLine = (length) => {
+    const line = "";
+    return line.padEnd(length, String.fromCharCode(0x2500));
+};
+
+const drawTopBorder = (maxLengths) => {
+    let paddedString = "";
```
Comment (QuickStyles) [inline box_it_script/BoxIt.js]: Unnecessary assignment. Just do:

```javascript
let paddedString = String.fromCharCode(0x250C);
```

File: box_it_script/BoxIt.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    paddedString = String.fromCharCode(0x250C);
```
Comment (QuickStyles) [inline box_it_script/BoxIt.js]: String.fromCharCode(0x250C); is very ambiguous.
For readability I would create a constant at the top of the file which has the value of that and then reference the variable where needed.
```javascript
const TOP_T_BAR = String.fromCharCode(0x250C);

paddedString = TOP_T_BAR;
```

File: box_it_script/BoxIt.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    rl.on('line', function(line) {
+        arr.push(line.split(','));
+    });
+
+    rl.on('close', function() {
+        console.log(boxIt(arr));
+    });
+} else {
+    console.log('Please provide a CSV file in the required format.');
+};
```
Comment (QuickStyles) [inline box_it_script/BoxIt.js]: Common practice is to leave an empty line at bottom of files

File: box_it_script/BoxIt.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        paddedString = paddedString.concat(drawBarsAround(tmpArrayStrings)).concat('\n');
+        if (index < arrayStringArrays.length-1) {
+            paddedString = paddedString.concat(drawMiddleBorder(maxLengths)).concat('\n');
+        };
+    };
+    return paddedString.concat(drawBottomBorder(maxLengths));
+};
+
+if (process.argv.length === 3 && process.argv[2].endsWith('.csv')) {
+    let fs = require('fs');
```
Comment (QuickStyles) [inline box_it_script/BoxIt.js]: Use `const` over `let` where ever possible

**Comment by Angadia** [comment]
Thank you Brandon for the feedback. I'll update this solution and subsequent solutions as per these suggestions.

**Pull request merged**